java/java.lsp.server/vscode/src/extension.ts

```diff
@@ -377,6 +377,107 @@ function getValueAfterPrefix(input: string | undefined, prefix: string): string
     return '';
 }
 
+class LineBufferingPseudoterminal implements vscode.Pseudoterminal {
+    private static instances = new Map<string, LineBufferingPseudoterminal>();
+
+    private writeEmitter = new vscode.EventEmitter<string>();
+    onDidWrite: vscode.Event<string> = this.writeEmitter.event;
+
+    private closeEmitter = new vscode.EventEmitter<void>();
+    onDidClose?: vscode.Event<void> = this.closeEmitter.event;
+
+    private buffer: string = ''; 
+    private isOpen = false;
+    private readonly name: string;
+    private terminal: vscode.Terminal | undefined;
+
+    private constructor(name: string) {
+        this.name = name;
+    }
+
+    open(): void {
+        this.isOpen = true;
+    }
+
+    close(): void {
+        this.isOpen = false;
+        this.closeEmitter.fire();
+    }
+
+    /**
+     * Accepts partial input strings and logs complete lines when they are formed.
+     * Also processes carriage returns (\r) to overwrite the current line.
+     * @param input The string input to the pseudoterminal.
+     */
+    public acceptInput(input: string): void {
+        if (!this.isOpen) {
+            return;
+        }
+
+        for (const char of input) {
+            if (char === '\n') {
+                // Process a newline: log the current buffer and reset it
+                this.logLine(this.buffer.trim());
+                this.buffer = '';
+            } else if (char === '\r') {
+                // Process a carriage return: log the current buffer on the same line
+                this.logInline(this.buffer.trim());
```

Comment:
Not sure about this: we have buffered content + instruction not to leave the line. So the result should be printed content from whenever the caret is at the moment + return at the start of line. 
 According to logInline comment, it will actially clear the current line + print the buffered content, leaving the caret at the end -- a reverse of the original instructions (content + CR)

Did I understand wrong ?

Reply:
`EventEmitter.fire(string)` always adds new line. We have to buffer whole lines because of it. There were unexpected line breaks with no buffering.